src/main/common/utils.js

```diff
@@ -347,3 +421,49 @@ gpii.app.getVolumeValue = function (browserWindow, messageChannel) {
         return defaultVolumeValue;
     }
 };
+
+/**
+ * Simple file function to check a path to the executable file
+ * @param {String} executablePath - path to executable file
+ * @return {Boolean} - returns `true` when the file exits and its executable
+ */
+gpii.app.checkExecutable = function (executablePath) {
+    try {
+        var fileProperties = fs.statSync(executablePath);
+        // Check that the file is executable
+        if (fileProperties.mode === parseInt("0100666", 8)) {
+            // returns true only if the file exists and its executable
+            return true;
+        } else {
+            fluid.log(fluid.logLevel.WARN, "checkExecutable: File is not executable - " + executablePath);
+        }
+    } catch (err) {
+        fluid.log(fluid.logLevel.WARN, "checkExecutable: Invalid or missing path - " + executablePath);
+    }
+    // returns false in any other case
+    return false;
+};
+
+/**
+ * Simple function to try to validate the url
+ * @param {String} url - browser's url
+ * @return {Boolean} - returns `true` when the url looks valid
+ */
+gpii.app.checkUrl = function (url) {
+    // validating the url
+    var pattern = new RegExp("^(https?:\\/\\/)?" + // protocol
```

Comment:
Give a source from where you copied this code :)

Reply:
bump - please give some attribution

Reply:
I've added the source before, I wonder what happened with it -- **re-added** now.
_(also second time typing this answer...)_